### ISSUE TITLE
FAILED External Storage [Driver: file.csi.azure.com] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]

### ISSUE THREAD

**Issue by andyzhangx** [kubernetes-sigs/azurefile-csi-driver#668]
**What happened**:

**What you expected to happen**:

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_azurefile-csi-driver/667/pull-azurefile-csi-driver-external-e2e/1396330790341578752/build-log.txt

```
External Storage [Driver: file.csi.azure.com]
[90m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/external/external.go:175[0m
  [Testpattern: Dynamic PV (default fs)] subPath
  [90m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/framework/testsuite.go:50[0m
    [91m[1mshould be able to unmount after the subpath directory is deleted [LinuxOnly] [Measurement][0m
    [90m/workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:444[0m

    [91mMay 23 05:32:43.534: while cleaning up resource
    Unexpected error:
        <errors.aggregate | len:2, cap:2>: [
            {
                s: "pod \"pod-subpath-test-dynamicpv-c58g\" was not deleted: timed out waiting for the condition",
            },
            [
                {
                    error: {
                        cause: {
                            s: "PersistentVolume pvc-771545c6-f9c6-4f2b-ad53-8ea4121ed9a4 still exists within 5m0s",
                        },
                        msg: "Persistent Volume pvc-771545c6-f9c6-4f2b-ad53-8ea4121ed9a4 not deleted by dynamic provisioner",
                    },
                    stack: [0x59c7ce5, 0x5a70648, 0x5a73ccf, 0x238dce3, 0x238d8fc, 0x238cc27, 0x2393bcf, 0x2393272, 0x2399511, 0x2399027, 0x2398817, 0x239ae06, 0x239ced8, 0x239cc2d, 0x5b2dd8c, 0x5b324eb, 0x20d3aaf, 0x201c281],
                },
            ],
        ]
        [pod "pod-subpath-test-dynamicpv-c58g" was not deleted: timed out waiting for the condition, Persistent Volume pvc-771545c6-f9c6-4f2b-ad53-8ea4121ed9a4 not deleted by dynamic provisioner: PersistentVolume pvc-771545c6-f9c6-4f2b-ad53-8ea4121ed9a4 still exists within 5m0s]
    occurred[0m

    /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:183
[90m------------------------------[0m
{"msg":"FAILED External Storage [Driver: file.csi.azure.com] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]","total":17,"completed":0,"skipped":804,"failed":1,"failures":["External Storage [Driver: file.csi.azure.com] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]"]}
```

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by andyzhangx** [comment]
related to issue: https://github.com/kubernetes/kubernetes/issues/97031

**Comment by Jiawei0227** [comment]
Has this get fixed?

**Comment by andyzhangx** [comment]
> Has this get fixed?

no, it's upstream issue, not related to this specific driver.

**Comment by andyzhangx** [comment]
this issue is fixed by https://github.com/kubernetes/kubernetes/pull/106906, and I am trying to cherry-pick to 1.23

**Comment by andyzhangx** [comment]
fixed by https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/907, this subPath issue was fixed in `1.21.9`, `1.22.6`, `1.23.2`
